File: LeetCodeTraining/LeetCodeTraining/summaryRanges.cpp
#include <iostream>
#include <vector>
#include <map>
using namespace std;

vector<string> summaryRanges(vector<int>& nums) {
    vector<string> answ;
    if (nums.size() == 1) {
        answ.push_back(to_string(nums[0]));
        return answ;
    }
    
    if (nums[0] == nums[1] - 1) {
        int start = nums[0];
        int end = nums[1];
        
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i + 1] - 1) {
                end = nums[i + 1];
            } else {
                if (end > start) {
                    answ.push_back(to_string(start) + "->" + to_string(end));
                    start = nums[i + 1];
                } else {
                    answ.push_back(to_string(start));
                }
                  
            }
        }
    } else {
        int start;
        int end;
        int j = 0;
        bool flag = true;
        while (flag) {
            if (nums[j] == nums[j + 1] - 1) {
                flag = false;
                int start = nums[j];
                int end = nums[j + 1];
            } else {
                answ.push_back(to_string(nums[j]));
            }
        }
        for (int i = j; i < nums.size(); i++) {
            if (nums[i] == nums[i + 1] - 1) {
                end = nums[i + 1];
            } else {
                answ.push_back(to_string(start) + "->" + to_string(end));
                start = nums[i + 1];
            }
        }
    }
    return answ;
}

File: LeetCodeTraining/LeetCodeTraining/summaryRanges.swift
import Foundation

func summaryRanges(_ nums: [Int]) -> [String] {
    guard !nums.isEmpty else {
        return []
    }
    
    var out: [String] = []
    var start: Int = nums[0];
    var prev = start;
    
    for i in 1..<nums.count {
        if (nums[i] > prev + 1) {
            addTwo(start, prev, &out)
            start = nums[i]
            prev = start
        } else {
            prev = nums[i]
        }
    }
    
    addTwo(start, prev, &out)
    
    return out
}


func addTwo(_ num1: Int, _ num2: Int, _ arr: inout [String]) {
    if (num1 != num2) {
        arr.append("\(num1)->\(num2)")
    } else {
        arr.append("\(num1)")
    }
}
